packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -3666,6 +3666,14 @@ export class Checker extends ParseTreeWalker {
         if (this._fileInfo.diagnosticRuleSet.reportShadowedImports === 'none') {
             return;
         }
+
+        // Skip this check for relative imports.
+        const nodeModule = node.nodeType === ParseNodeType.ImportFromAs ? (node.parent as any)?.module : node.module;
+        if (nodeModule?.importInfo?.isRelative) {
```

Comment:
See above. Please don't access importInfo directly without going through the accessor function. And there's no need to use this here; you should use the syntax information (`module.leadingDots.length > 0`) instead.